pkg/controller/elasticsearch/certificates/transport/csr_test.go

```diff
@@ -97,6 +98,32 @@ func Test_buildGeneralNames(t *testing.T) {
 				{IPAddress: net.ParseIP("127.0.0.1").To4()},
 			},
 		},
+		{
+			name: "optional user provided SANs",
+			args: args{
+				cluster: func() esv1.Elasticsearch {
+					es := testES
+					es.Spec.Transport.TLS.SubjectAlternativeNames = []commonv1.SubjectAlternativeName{
+						{
+							DNS: "my-custom-domain",
+							IP:  "111.222.333.444",
```

Comment:
I would argue this is an unexpected way to use the API and not the one we document.

What we test here:

```
- ip: 1.2.3.4
  dns: abcd.com
```

vs what we document:

```
- ip: 1.2.3.4
- dns: abcd.com
```

Both are valid though, let's make sure our unit test encapsulate them (i.e. add more entries to that test array 😄)?

Reply:
Good point.